### ISSUE TITLE
Support keybinds with modifiers

### ISSUE THREAD

**Issue by bovine3dom** [cmcaine/tridactyl#41]
e.g, ^U/^D.

This would involve reverting 37119e510a82755c4e57774476bfa2870555c1c4, which drops all keypresses containing modifiers.

**Comment by dumblob** [comment]
Let me note, that there any key can behave like a modifier - it's dependent on OS and the keyboard layout (yep, I'm one of those, who has keys remapped already in a DLL under Windows and therefore I can use e.g. the physical key `Tab` as `Right Alt`).

Why I'm writing this? Because reverting the above mentioned commit would result in support just for, and I quote ` || keyevent.ctrlKey || keyevent.altKey`.

**Comment by mMontu** [comment]
A possible workaround, at least for Windows users, is to map through [AutoHotKey](https://www.autohotkey.com/):

    ;; --------------------------------------------------------
    ;; -- Mapppings for Tridactyl --
    ;; --------------------------------------------------------
    #ifWinActive .*Firefox
       Capslock::Esc
       Ctrl & n::Send gt
       Ctrl & p::Send gT
    #ifWinActive

**Comment by wnereiz** [comment]
@mMontu There's [autokey](https://github.com/autokey/autokey) on Linux

**Comment by bovine3dom** [comment]
@cmcaine and I were planning to actually fix this this weekend, so you might not need to use your work around for long :)

**Comment by mMontu** [comment]
@wnereiz nice, I've never heard about it - have you tried it? Is it close to AHK?

@bovine3dom great news!!! And thanks for this amazing project, it is really good for something in "in a pretty early stage of development".

**Comment by xerebz** [comment]
Any one actually have a workaround for this on Linux? Can you provide an autokey example?

**Comment by wnereiz** [comment]
@mMontu I didn't try it myself. I'm still waiting the support of modifiers.

There's another "dirty" way to solve this problem - to change hardcoded key binding in the firefox source code. The downside is you have to rebuild every time when there is a new update. With the help of [open build service](https://github.com/openSUSE/obs-build), eg. [openSUSE build service](https://build.opensuse.org/), it is possible to trigger this process automatically. And you could even have your own repository.

But it is still the "dirty" way, although it could solve the problem theoretically. I will not do that until there is no choices. Now, the modifier support is on the way. So, I am waiting.

**Comment by timokau** [comment]
Has there been any update on this? I don't want to nag, just to make sure this issue wasn't forgotten (and maybe the problem already resolved in master?)

**Comment by bovine3dom** [comment]
This is going to sound like a joke, but @cmcaine was hoping to get it done this evening.

**Comment by cmcaine** [comment]
I actually have it working on my machine *right now* :)

I think we should make it a premium only option. Tridactyl Authentic Premium Edition costs one imperial pound of 24 carat gold, deliverable to our Manchester address.

**Comment by bovine3dom** [comment]
Surely you mean Emacs edition?

**Comment by timokau** [comment]
This should close https://github.com/cmcaine/tridactyl/issues/304.

Is it possible to `bind <C-n> tabnext`? When I try, `<C-n>` still opens a new window. `bind <C-p> tabprev` works (and does not open the print dialog).

**Comment by bovine3dom** [comment]
No, sorry. Some binds are reserved by Firefox, and it looks like C-n is one of them. You could try using A-n instead?

We'll probably dust off the keyboard API we were working on at some point which hopefully would let us get around that.

**Comment by timokau** [comment]
Okay, I'll just have to get used to a different binding then. Thanks.

**Comment by thraen** [comment]
I need to rebind <c-i>,<c-p> and <c-n>. does anyone have a tip on where to turn to at mozilla to nag about that?_

**Comment by bovine3dom** [comment]
@thraen `<c-i>` should already work. Possible that you are colliding with one of our own default binds. Please check `viewconfig` to see if you have multiple similar ones (although I thought we'd fixed that issue).

As for the others, I dunno. I haven't learnt the art of navigating bugzilla. You can try your luck with https://bugzilla.mozilla.org/ but please only "vote" by clicking the button and file novel issues. Posting comments in support of things will probably just annoy the moderators there. (I don't know whether "user stories" fall under that umbrella).
